### ISSUE TITLE
`propose` does not push branch to remote before redirecting to web host

### ISSUE THREAD

**Issue by tranhl** [git-town/git-town#4164]
### Steps to reproduce

1. `git-town hack test-branch`: Create a new prototype branch
2. Make changes
3. `git commit`
4. `git-town propose`: Create a proposal for these changes
5. Once redirected to GitHub, we find that there are no changes to create a PR with

![image](https://github.com/user-attachments/assets/f50493b5-cd1b-4397-b4cd-dd08ffe21908)

### Command output

```shell
I've redacted the remote refs from the verbose output to protect company data.


Git Town 16.4.1

(verbose) git version
git version 2.46.0

(verbose) git rev-parse --show-toplevel
/Users/<redacted>/Projects/monorepo

(verbose) git config -lz --includes --global
advice.addignoredfile
falsecore.editor
nvimcredential.https://gist.github.com.helper
/nix/store/fy4lwzzzxcyz3z4m22nsiywxnyla0fnz-gh-2.58.0/bin/gh auth git-credentialcredential.https://github.com.helper
/nix/store/fy4lwzzzxcyz3z4m22nsiywxnyla0fnz-gh-2.58.0/bin/gh auth git-credentialfilter.lfs.clean
git-lfs clean -- %ffilter.lfs.process
git-lfs filter-processfilter.lfs.required
truefilter.lfs.smudge
git-lfs smudge -- %fgit-town.github-token
<redacted>git-town.main-branch
maingit-town.ship-delete-tracking-branch
falseinit.defaultbranch
mainuser.email
<redacted>
user.name
Long Tran
(verbose) git config -lz --includes --local
core.repositoryformatversion
0core.filemode
truecore.bare
falsecore.logallrefupdates
truecore.ignorecase
truecore.precomposeunicode
truecore.hookspath
.husky/_submodule.active
.remote.origin.url
git@github.com:<redacted>/monorepo.gitremote.origin.fetch
+refs/heads/*:refs/remotes/origin/*branch.main.remote
originbranch.main.merge
origin/maingit-town-branch.persister.parent
mainbranch.persister.github-pr-base-branch
<redacted>#monorepo#mainbranch.persister.vscode-merge-base
origin/mainbranch.persister.remote
originbranch.persister.merge
refs/heads/persister
(verbose) git status --long --ignore-submodules
On branch persister
Your branch is up to date with 'origin/persister'.

nothing to commit, working tree clean

(verbose) git remote
origin

(verbose) git rev-parse --abbrev-ref HEAD
persister

[persister] git fetch --prune --no-tags

(verbose) git stash list

(verbose) git branch -vva --sort=refname
  main                                                                                          83ec24a72 [origin/main] <redacted>
* persister                                                                                     7ab605f04 [origin/persister] 🐛 Persisted cache not preventing refetch
  remotes/origin/main                                                                           83ec24a72 <redacted>
  remotes/origin/persister                                                                      7ab605f04 🐛 Persisted cache not preventing refetch

(verbose) git rev-parse --verify --abbrev-ref @{-1}
main

(verbose) git remote get-url origin
git@github.com:<redacted>/monorepo.git

Looking for proposal online ... none

(verbose) git log main..persister --format=%h
7ab605f04

(verbose) git log --format=%B -n 1 7ab605f04
🐛 Persisted cache not preventing refetch


[persister] git checkout main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.

[main] git rebase origin/main --no-update-refs
Current branch main is up to date.

(verbose) git rev-list --left-right main...origin/main

[main] git checkout persister
Switched to branch 'persister'
Your branch is up to date with 'origin/persister'.

[persister] git merge --no-edit --ff origin/persister
Already up to date.

[persister] git merge --no-edit --ff main
Already up to date.

(verbose) git config git-town.prototype-branches query-persistence

(verbose) git show-ref --verify --quiet refs/heads/main

(verbose) which wsl-open

(verbose) which garcon-url-handler

(verbose) which xdg-open

(verbose) which open
/usr/bin/open

open https://github.com/<redacted>/monorepo/compare/persister?expand=1

(verbose) git branch -vva --sort=refname
  main                                                                                          83ec24a72 [origin/main] <redacted>
* persister                                                                                     7ab605f04 [origin/persister] 🐛 Persisted cache not preventing refetch
  remotes/origin/main                                                                           83ec24a72 <redacted>
  remotes/origin/persister                                                                      7ab605f04 🐛 Persisted cache not preventing refetch

(verbose) git config -lz --includes --global
core.repositoryformatversion
0core.filemode
truecore.bare
falsecore.logallrefupdates
truecore.ignorecase
truecore.precomposeunicode
truecore.hookspath
.husky/_submodule.active
.remote.origin.url
git@github.com:<redacted>/monorepo.gitremote.origin.fetch
+refs/heads/*:refs/remotes/origin/*branch.main.remote
originbranch.main.merge
origin/maingit-town-branch.persister.parent
mainbranch.persister.github-pr-base-branch
<redacted>#monorepo#mainbranch.persister.vscode-merge-base
origin/mainbranch.persister.remote
originbranch.persister.merge
refs/heads/persister
(verbose) git stash list

Ran 31 shell commands.
branch "persister" is no longer a prototype branch
```
```


### Additional information

Git Town config:

```toml
push-hook = true
create-prototype-branches = true
push-new-branches = false
ship-delete-tracking-branch = false
sync-before-ship = true
sync-upstream = true
ship-strategy = "api"
sync-tags = false

[branches]
main = "main"
perennial-regex = "^release-.*$"

[hosting]
platform = "github"

[sync-strategy]
feature-branches = "merge"
perennial-branches = "rebase"
```

**Comment by kevgo** [comment]
Thanks for reporting this so detailed. I think this is a duplicate of https://github.com/git-town/git-town/issues/4109 which is already fixed, but that fix is not released yet. Time to make another release soon!

**Comment by tranhl** [comment]
No problems! It's great to hear that a fix is already on the way 😁

**Comment by kevgo** [comment]
This is now shipped in [Git Town 16.5](https://github.com/git-town/git-town/releases/tag/v16.5.0).
